### DISCUSSION TITLE
postcss error on build

### DISCUSSION THREAD

**Discussion by floflor** [postcss/postcss#1656] [Q&A]
Hi everyone ! I'm having a weird error when building my app on Jenkins. This comes from postcss and I can't find any information online. 

**Console output:** 

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/parser' is not defined by "exports" in /app/node_modules/next/node_modules/postcss/package.json

![image](https://user-images.githubusercontent.com/32499564/139960042-29cc4668-b3d3-4c75-a933-1addf6c2e847.png)

**PostCss version:** 8.2.4

Any help would be appreciated ! 🙏🏼

**Comment by ai** [comment]
Very strange, we have this export https://github.com/postcss/postcss/blob/main/package.json#L26

What did you do before getting this error?

**Reply by floflor** [reply]
We noticed this error yesterday, we had an instance running perfectly on thursday, then we pushed three prs to development , just css corrections and some logic fixes, but nothing related to postcss or package config.

**Reply by ai** [reply]
PostCSS 7.x on new Node.js 17 have exactly the same error.

Try to check your `npm ls` or lock file for old PostCSS version.

**Reply by floflor** [reply]
Hey Andrey , we updated node version to 12 on docker file and this fixed the issue, thank you so much for your help.

**Reply by ai** [reply]
Good to know. I just recommend update further to Node.js 14 or 16 sober Node.js 12 is old and will be deprecated next year (so you will again have unknown errors).

**Reply by patra0o** [reply]
Hi, I'm on Node.js 18 and I do not understand why I get build errors. I'm not even using PostCSS directly in my project.
![image](https://user-images.githubusercontent.com/67415279/214013890-5c65c024-4f56-47d8-a72d-2988fff4736e.png)
